src/realm/object-store/c_api/notifications.cpp

```diff
@@ -238,6 +259,40 @@ RLM_API void realm_collection_changes_get_ranges(
     }
 }
 
+RLM_API void realm_dictionary_get_changes(const realm_dictionary_changes_t* changes, size_t* out_deletions_size,
+                                          size_t* out_insertion_size, size_t* out_modification_size)
+{
+    if (out_deletions_size)
+        *out_deletions_size = changes->deletions.size();
+    if (out_insertion_size)
+        *out_insertion_size = changes->insertions.size();
+    if (out_modification_size)
+        *out_modification_size = changes->modifications.size();
+}
+
+RLM_API void realm_dictionary_get_changed_keys(const realm_dictionary_changes_t* changes,
+                                               realm_value_t* deletion_keys, size_t* deletions_size,
+                                               realm_value_t* insertion_keys, size_t* insertions_size,
+                                               realm_value_t* modification_keys, size_t* modifications_size)
+{
+    auto fill = [](const auto& collection, realm_value_t* out, size_t* n) {
+        if (!out || !n)
+            return;
+        if (collection.size() == 0 || *n < collection.size()) {
+            *n = 0;
+            return;
+        }
+        size_t i = 0;
+        for (auto val : collection)
+            out[i++] = to_capi(val);
+        *n = i;
+    };
+
+    fill(changes->deletions, deletion_keys, deletions_size);
```

Comment:
This could be simpler (see previous suggestion).
```suggestion
    fill(changes->deletions, deletion_keys, changes->deletions.size());
```

Reply:
I don't think it can be simplified, because we have 2 calls one for retrieving the length of the collection and one for filling it with the values, it could potentially happen that some change can sneak in the middle, but since the array is allocated by the SDK we can only use the memory we were provided. This is why the size of each single collection is passed as output parameter.